src/java.base/share/classes/java/util/concurrent/ConcurrentSkipListSet.java

```diff
@@ -527,12 +528,11 @@ public Spliterator<E> spliterator() {
 
     /** Initializes map field; for use in clone. */
     private void setMap(ConcurrentNavigableMap<E,Object> map) {
-        try {
-            Field mapField = ConcurrentSkipListSet.class.getDeclaredField("m");
-            mapField.setAccessible(true);
-            mapField.set(this, map);
-        } catch (IllegalAccessException | NoSuchFieldException e) {
-            throw new Error(e);
-        }
+        final Unsafe U = Unsafe.getUnsafe();
+        U.putReference(
+            this,
+            U.objectFieldOffset(ConcurrentSkipListSet.class, "m"),
```

Comment:
Would it make sense to compute the offset once and for all and put it in a `static final long` field?

Reply:
Yeah, I originally did that, but the following patch is the "smallest change".
Given that the "original code" obtained the Field instance each call, this is still likely a performance improvement.

Reply:
Fine. The main effect of this change is to nearly revert to the pre-VarHandle version of this code.